### PR TITLE
Various eval fixes with `allowAliases = false`

### PR DESCRIPTION
**Pull request by infinisil**
## Description of changes

This fixes evaluation of various attributes with `allowAliases = false`, sometimes by removing them if they shouldn't exist at all with `allowAliases = false`:

- chia
- chia-dev-tools
- chia-plotter
- dotnet-sdk_2
- dotnet-sdk_3
- dotnet-sdk_5
- gofumpt
- gruvbox-plus-icons
- mod_wsgi
- nextcloud25
- nextcloud25Packages
- nix-repl
- teams
- wtk

## Things done

- Tested evaluation of these attributes with `nix-instantiate --eval --arg config '{ allowAliases = false; }' -A <attr>`. They either don't exist or succeed

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/teams/default.nix [hunk 1]
```diff
@@ -26,37 +26,33 @@ let
...
-      runHook postInstall
-    '';
+  src = fetchurl {
+    url = "https://statics.teams.cdn.office.net/production-osx/${versions.darwin}/Teams_osx.pkg";
+    hash = hashes.darwin;
   };
-in
-if stdenv.isDarwin
-then darwin
-else throw "Teams app for Linux has been removed as it is unmaintained by upstream. (2023-09-29)"
```
Comment (eureka-cpu) [inline pkgs/applications/networking/instant-messengers/teams/default.nix:62]: Is it unreasonable to still include this error message for Linux users?

File: pkgs/servers/nextcloud/default.nix [hunk 2]
```diff
@@ -42,18 +42,6 @@ let
...
-    by upstream in 2023-10. Please upgrade to at least Nextcloud v26 by declaring
-
-        services.nextcloud.package = pkgs.nextcloud26;
-
-    in your NixOS config.
-
-    WARNING: if you were on Nextcloud 24 you have to upgrade to Nextcloud 25
-    first on 23.05 because Nextcloud doesn't support upgrades across multiple major versions!
-  '';
-
```
Comment (eureka-cpu) [inline pkgs/servers/nextcloud/default.nix:56]: This is still fairly recent. Can we not remove this?

File: pkgs/by-name/gr/gruvbox-plus-icons/package.nix [hunk 3]
```diff
@@ -21,7 +21,7 @@ stdenvNoCC.mkDerivation {
 
   nativeBuildInputs = [ gtk3 ];
 
-  propagatedBuildInputs = [ breeze-icons gnome-icon-theme hicolor-icon-theme ];
+  propagatedBuildInputs = [ plasma5Packages.breeze-icons gnome-icon-theme hicolor-icon-theme ];
```
Comment (eureka-cpu) [inline pkgs/by-name/gr/gruvbox-plus-icons/package.nix:24]: Are these changes necessary? They seem to bringing more than just the icons into scope for an icon package.

File: pkgs/top-level/aliases.nix [hunk 4]
```diff
@@ -629,10 +647,26 @@ mapAliases ({
...
+    Nextcloud v25 has been removed from `nixpkgs` as the support for is dropped
+    by upstream in 2023-10. Please upgrade to at least Nextcloud v26 by declaring
+
+        services.nextcloud.package = pkgs.nextcloud26;
+
+    in your NixOS config.
+
+    WARNING: if you were on Nextcloud 24 you have to upgrade to Nextcloud 25
+    first on 23.05 because Nextcloud doesn't support upgrades across multiple major versions!
+  ''; # Added 2023-10-13
```
Comment (eureka-cpu) [inline pkgs/top-level/aliases.nix:660]: Disregard my previous comment about the message being removed.

**Review by eureka-cpu — COMMENTED** [review]
It seems there are some changes here that can be viewed as unrelated, ie the changes to the icon package. If it's not too much trouble, can you group the commits separately and add closing keywords for corresponding issues? 

I'm fairly new to the Nix language btw, so feel free to ignore my review if you don't find it helpful 🙂

**Review by katexochen — APPROVED** [review]
Changes lgtm, thanks. :)
I guess it's to expensive to check this on every PR?

File: pkgs/applications/networking/instant-messengers/teams/default.nix (hunk 1 above)
Comment (infinisil) [reply pkgs/applications/networking/instant-messengers/teams/default.nix:62]: There's no good infrastructure to have that easily, and I think the default error for unsupported platforms is good enough:
```
error: Package ‘teams-1.6.00.4464’ in /home/tweagysil/src/nixpkgs/pr/pkgs/applications/networking/instant-messengers/teams/default.nix:18 is not available on the requested hostPlatform:
         hostPlatform.config = "x86_64-unknown-linux-gnu"
         package.meta.platforms = [
           "x86_64-darwin"
           "aarch64-darwin"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.
```

File: pkgs/servers/nextcloud/default.nix (hunk 2 above)
Comment (infinisil) [reply pkgs/servers/nextcloud/default.nix:56]: Not sure what you mean. I just moved this part around while keeping the same error message, since I know close to nothing about all of these packages :laughing:

File: pkgs/by-name/gr/gruvbox-plus-icons/package.nix (hunk 3 above)
Comment (infinisil) [reply pkgs/by-name/gr/gruvbox-plus-icons/package.nix:24]: Yeah this is necessary because on master, this fails:
```
$ nix-instantiate -A gruvbox-plus-icons --arg config '{ allowAliases = false; }'
error:
       … while calling the 'throw' builtin

         at /home/tweagysil/src/nixpkgs/pr/lib/customisation.nix:206:13:

          205|        then makeOverridable f allArgs
          206|        else throw "lib.customisation.callPackageWith: ${error}";
             |             ^
          207|

       error: lib.customisation.callPackageWith: Function called without required argument "breeze-icons" at /home/tweagysil/src/nixpkgs/pr/pkgs/by-name/gr/gruvbox-plus-icons/package.nix:6
```

This is because `breeze-icons` is an alias. We can look into `aliases.nix` to see where it comes from: https://github.com/NixOS/nixpkgs/blob/989df07dc54ec214b324a720b5bdf056c3132a3e/pkgs/top-level/aliases.nix#L1087-L1089

So the change here just inlines the alias, therefore not using the alias anymore

File: pkgs/servers/nextcloud/default.nix (hunk 2 above)
Comment (infinisil) [reply pkgs/servers/nextcloud/default.nix:56]: Ah just saw https://github.com/NixOS/nixpkgs/pull/276112#discussion_r1435352321 haha

**Comment by infinisil** [comment]
>  I guess it's to expensive to check this on every PR?

It's actually very cheap, under 10 seconds on my machine! Here's how to see all failing attributes on `x86_64-linux`:

```
❯ nix repl lib
warning: future versions of Nix will require using `--file` to load a file
Welcome to Nix 2.18.1. Type :? for help.

Loading installable ''...
Added 422 variables.
nix-repl> attrNames (filterAttrs (name: value: ! (builtins.tryEval value).success) (import ./. { config.allowAliases = false; system = "x86_64-linux"; }))
[ "AAAAAASomeThingsFailToEvaluate" "androidsdk_9_0" "gams" "gccWithoutTargetLibc" "libcCross" "mathematica10" "mathematica11" "mathematica9" "neoload" "pkgsx86_64Darwin" "sc2-headless" "yabai" ]
```

There's not a lot left, and a bunch will also be fixed with https://github.com/NixOS/nixpkgs/pull/276120.

However, having taken a look at the rest of these, there are some kind of tricky cases which I'm not sure yet how to handle. Furthermore, the list is much longer if you use e.g. `x86_64-darwin` instead, or even some obscure platform. Then the question is whether this should be checked for all platforms somehow. And then there's also cross compilation, which probably has a ton more failures.

So overall, this PR is just a small step in the bigger picture :)

File: pkgs/by-name/gr/gruvbox-plus-icons/package.nix (hunk 3 above)
Comment (eureka-cpu) [reply pkgs/by-name/gr/gruvbox-plus-icons/package.nix:24]: Ah I see

**Review by eureka-cpu — APPROVED** [review]

**Comment by katexochen** [comment]
Thanks for the nice explanations (also on the review comments) @infinisil. It's great to get such background, I learned a lot reviewing this PR! :blush:

**Comment by RGBCube** [comment]
LGTM as well

**Review by liff — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -26695,20 +26692,6 @@ with pkgs;
 
   mkchromecast = libsForQt5.callPackage ../applications/networking/mkchromecast { };
 
-  # Backwards compatibility.
-  mod_dnssd = apacheHttpdPackages.mod_dnssd;
```
Comment (jtojnar) [inline pkgs/top-level/all-packages.nix:26699]: This broke building GNOME with aliases disabled, fixed in https://github.com/NixOS/nixpkgs/pull/278744

Comment (lilyinstarlight) [reply pkgs/top-level/all-packages.nix:26699]: Ofborg didn't catch it when calculating outpaths because of https://github.com/NixOS/nixpkgs/pull/271123#discussion_r1442134594 as well, by the way (I've been investigating this)

Comment (infinisil) [reply pkgs/top-level/all-packages.nix:26699]: I'm working on a fix

Comment (infinisil) [reply pkgs/top-level/all-packages.nix:26699]: https://github.com/NixOS/nixpkgs/pull/278777

**Comment by nyabinary** [comment]
You readded Microsoft Teams with this btw (accidentally?)
